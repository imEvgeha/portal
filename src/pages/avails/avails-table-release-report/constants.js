export const CREATE_REPORT = 'Create Report';
export const NEW_RELEASE_REPORT = 'Report';
export const START_YEAR = 1970;
export const END_YEAR = 2030;
export const MOCK_YEAR = 2099;
export const ERROR_MESSAGE = 'No records found';
export const MONTHS = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
];
