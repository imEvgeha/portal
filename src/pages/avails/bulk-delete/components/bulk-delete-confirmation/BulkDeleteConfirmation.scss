@import '../../../../../styles/atlaskit/mixins';

.nexus-c-bulk-delete-confirmation {
    height: 100%;
    width: 100%;
}

.nexus-c-bulk-delete-confirmation__message {
    @include margin(0, 0, 5, 0);
}

.nexus-c-bulk-delete-confirmation__container {
    @include addSpacing('max-height', 100);
    overflow: auto;
}
