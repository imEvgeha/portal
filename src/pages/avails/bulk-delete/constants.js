export const HEADER = {
    cells: [
        {
            key: 1,
            content: 'Linked Rights',
            width: 30,
            shouldTruncate: true,
        },
        {
            key: 2,
            content: 'Right ID',
            width: 15,
        },
        {
            key: 3,
            content: 'Type',
            width: 10,
        },
        {
            key: 4,
            content: 'Original Right',
            width: 25,
        },
        {
            key: 5,
            content: 'Source Right',
            width: 20,
        },
    ],
};

export const BULK_DELETE_WARNING_MSG = num => `Permanently delete ${num} Rights?`;
export const BULK_DELETE_REMAINING_MSG = (num1, num2) =>
    `${num1} Rights have been deleted. Deleting some of the remaining ${num2} Rights will also delete their linked TPR/Bonus Rights. Please review.`;
export const BULK_DELETE_LINKED_RIGHT_MSG = 'The following linked TPR/Bonus rights will also be marked as deleted:';
export const SELECTED_RIGHTS = 'Selected Rights';
export const AFFECTED_RIGHTS = 'Affected Rights';
export const BULK_DELETE_CONTINUE_MSG = 'How would you like to continue?';
export const BULK_DELETE_BTN_DELETE = 'Mark as Deleted';
export const BULK_DELETE_BTN_CANCEL = 'Cancel';

export const RIGHT_ID = 'Right ID';
export const TITLE = 'Title';
export const LINKED_RIGHT_ID = 'Linked Right ID';
export const TPR_ORIGINAL_RIGHTS = 'TPR Original Rights';
export const BONUS_SOURCE_RIGHT = 'Bonus Source Right';
