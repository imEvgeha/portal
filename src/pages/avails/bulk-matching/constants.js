export const TITLE_MATCHING_MSG = 'Select titles from the repositories that match the incoming right or declare a ';
export const TITLE_MATCHING_REVIEW_HEADER = 'Title Matching Review';
export const RIGHT_TABS = {
    SELECTED: 'Selected Rights',
    AFFECTED: 'Affected Rights',
    BONUS_RIGHTS: 'Existing Bonus Rights',
};
