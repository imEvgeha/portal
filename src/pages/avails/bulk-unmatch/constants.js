export const BULK_UNMATCH_TITLE = 'Bulk Unmatch Rights Review';
export const BULK_UNMATCH_WARNING = 'Warning: Please be aware you need to validate title fields.';
export const BULK_UNMATCH_CONFIRM_BTN = 'Confirm';
export const BULK_UNMATCH_CANCEL_BTN = 'Cancel';

