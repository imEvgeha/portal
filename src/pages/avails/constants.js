export default {
    PAGE_SIZE: 100,
    sortParams: [{id: 'received', desc: 'true'}],
    AVAIL_HISTORY_ID: 'availHistoryIds',
    INGEST_HISTORY_ATTACHMENT_ID: 'ingestHistoryAttachmentId',
    INITIAL_SORT: {colId: 'updatedAt', sort: 'desc'},
};
