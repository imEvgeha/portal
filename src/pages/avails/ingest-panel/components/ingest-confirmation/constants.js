export const INGEST_ASSIGN_MESSAGE = 'You have assigned your upload with:';
export const INGEST_UPLOAD_CONTINUE_MSG = 'Do you wish to continue?';
export const INGEST_UPLOAD_CANCEL_BTN = 'Cancel';
export const INGEST_UPLOAD_CONTINUE_BTN = 'Continue';
