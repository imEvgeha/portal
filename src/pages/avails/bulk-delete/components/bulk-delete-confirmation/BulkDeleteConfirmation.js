import React from 'react';
import PropTypes from 'prop-types';
import {BULK_DELETE_WARNING_MSG} from '../../constants';
import BulkDeleteActions from '../bulk-delete-actions/BulkDeleteActions';
import './BulkDeleteConfirmation.scss';

export const BulkDeleteConfirmation = ({onClose, onSubmit, rightsCount}) => {
    return (
        <div className="nexus-c-bulk-delete-confirmation">
            <div className="nexus-c-bulk-delete-confirmation__container">
                <div className="nexus-c-bulk-delete-confirmation__message">{BULK_DELETE_WARNING_MSG(rightsCount)}</div>
            </div>
            <BulkDeleteActions onClose={onClose} onSubmit={onSubmit} rightsDeletionCount={rightsCount} />
        </div>
    );
};

BulkDeleteConfirmation.propTypes = {
    onClose: PropTypes.func,
    onSubmit: PropTypes.func,
    rightsCount: PropTypes.number,
};

BulkDeleteConfirmation.defaultProps = {
    onClose: () => null,
    onSubmit: () => null,
    rightsCount: 0,
};

export default BulkDeleteConfirmation;
