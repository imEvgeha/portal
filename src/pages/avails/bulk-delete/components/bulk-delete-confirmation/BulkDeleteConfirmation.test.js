import React from 'react';
import {shallow} from 'enzyme';
import {BULK_DELETE_WARNING_MSG} from '../../constants';
import BulkDeleteConfirmation from './BulkDeleteConfirmation';

describe('BulkDeleteConfirmation', () => {
    let wrapper = null;

    const props = {
        rightsCount: 3,
    };

    beforeEach(() => {
        wrapper = shallow(<BulkDeleteConfirmation {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders bulk delete confirmation message', () => {
        expect(wrapper.find('.nexus-c-bulk-delete-confirmation__message').text()).toEqual(
            `${BULK_DELETE_WARNING_MSG(props.rightsCount)}`
        );
    });

    it('renders Delete and Cancel buttons', () => {
        expect(wrapper.find('BulkDeleteActions')).toHaveLength(1);
    });
});
