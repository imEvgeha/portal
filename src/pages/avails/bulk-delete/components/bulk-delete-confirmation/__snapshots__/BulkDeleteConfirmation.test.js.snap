// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkDeleteConfirmation should match snapshot 1`] = `
<div
  className="nexus-c-bulk-delete-confirmation"
>
  <div
    className="nexus-c-bulk-delete-confirmation__container"
  >
    <div
      className="nexus-c-bulk-delete-confirmation__message"
    >
      Permanently delete 3 Rights?
    </div>
  </div>
  <BulkDeleteActions
    isDisabled={false}
    isLoading={false}
    onClose={[Function]}
    onSubmit={[Function]}
    rightsDeletionCount={3}
  />
</div>
`;
